### PR TITLE
 Add docs for `useConnect` and update `connect`

### PR DESCRIPTION
**Pull request by DAreRodz**
Documentation for https://github.com/frontity/frontity/pull/404

### CODE REVIEW

**Comment by DAreRodz** [comment]
Some code was autoformatted, I guess that's not a problem for Gitbook, right?

cc: @juanmaguitar, @mburridge

File: docs/api-reference-1/frontity.md [hunk 1]
```diff
@@ -66,22 +67,27 @@ Frontity exports `fetch` and `URL` with the same API they have in the browser, b
 #### Syntax
 
 ```javascript
-ConnectedComponent = connect(Component);
+ConnectedComponent = connect(Component, options?);
 ```
 
 It's a function that receives a React component an returns the same component but connected to the Frontity state, actions and libraries. Any instance of that component will receive three new props: `state`, `actions` and `libraries`, allowing the component to read the state, manipulate it through actions or use any code other packages have exposed in libraries. Also, that instance will re-render automatically whenever the value of `state` which the component is using is changed.
 
+If you don't want to inject new props in your components, you can use the `injectProps` option set to `false`. That will make your component reactive to changes in the state but without passing new props. For these components to access the state use the [`useConnect`](frontity.md#useConnect) hook.
```
Comment (michalczaplinski) [inline docs/api-reference-1/frontity.md:75]: Could you clarify what you mean by "without passing new props" ? Which new props is it referring to? Does that mean that the value of the `state` props is never going to be updated in components with `{injectProps: false}` ? 

Edit: I see that you clarify it below, but I would put it here as well 🙂

File: docs/api-reference-1/frontity.md [hunk 2]
```diff
@@ -66,22 +67,27 @@ Frontity exports `fetch` and `URL` with the same API they have in the browser, b
...
 
-* `Component`: a React component.
+- `Component`: a React component.
+- `options` (optional): object with the following properties:
+  - `injectProps`: Boolean. If `false`, the `state`, `actions` and `libraries` won't be passed as props to the component. Default is `true`.
 
 #### Return value
 
-* The same component but connected to `state`, `actions` and `libraries`.
+- Same component connected to the Frontity state.
```
Comment (michalczaplinski) [inline docs/api-reference-1/frontity.md]: The same component as passed as the first argument, but connected to the Frontity state.

File: docs/api-reference-1/frontity.md (hunk 1 above)
Comment (DAreRodz) [reply docs/api-reference-1/frontity.md:75]: I'll do that, thanks Michal :+1:

**Comment by DAreRodz** [comment]
I slightly changed the docs, I hope it's more clear now. @michalczaplinski, @juanmaguitar could you review this?

**Comment by DAreRodz** [comment]
Thanks!

**Comment by luisherranz** [comment]
Hey guys, just to clarify this, people don't need to use `{ injectProps: false }` to use `useConnect`. Only they want to avoid the props. So it's optional.

And I think it should not be recommended, except for the case where you want to pass down props to a HTML tag, like:

```js
const Input = ({ state, ...restProps}) => {
   // Do something with `state`.

  return <input {...restProps} />
};

export default connect(Input);
```

Because in this case, `input` don't want that `actions` and `libraries` end up in the DOM.

```js
const Input = (props) => {
    const { state } = useConnect();
   // Do something with `state`.

  return <input {...props} />
};

// In this case, avoid injecting `state`, `actions` and `libraries` so they are not in `...props`.
export default connect(Input, { injectProps: false });
```

The rest of the time, people should use just `connect`. This is perfectly fine:

```js
const Input = ({ name, type }) => {
    const { state } = useConnect();
   // Do something with `state`.

  return <input name={name} type={type} />
};

export default connect(Input);
```

@juanmaguitar could you please clarifying this in the docs?

**Comment by juanmaguitar** [comment]
@luisherranz @DAreRodz clarification addded → https://github.com/frontity/gitbook-docs/pull/71

**Pull request merged**